docs/7.x/testplan.md

```diff
@@ -429,17 +431,39 @@ root$ gravity resource create cloud-config.yaml --confirm -m
 
 This scenario tests garbage collection on a cluster.
 
-Prerequisites: multi-node cluster with at least 1 regular node. Regular node is necessary to test both master and regular node update paths.
+Prerequisites: multi-node cluster with at least 1 node `--role=knode` and 1 `--role=node|master`.
+Having both `knode` and `node|master` is necessary to test both master and regular node update paths.
 
-Install a previous LTS version, upgrade to the latest version.
+Install a previous LTS version. Gather baseline pre-upgrade state:
 
-After upgrade execute `gravity gc` on the cluster.
+- [ ] `sudo du -sh /var/lib/gravity/{local/packages,site/packages,planet/registry}`
 
-- [ ] Verify the operation completes successfully.
- - [ ] Verify that packages from the previous installation have been removed locally.
- - [ ] Verify that packages from the previous installation have been removed from cluster package storage.
- - [ ] Verify that packages from the current installation are still present.
- - [ ] Tentative: Verify that application packages from remote clusters are still present.
+Upgrade to the release under test. Gather pre-garbage-collection state.
+
+- [ ] `sudo du -sh /var/lib/gravity/{local/packages,site/packages,planet/registry}`
+- [ ] List all images/tags in the registry
+ - `sudo find /var/lib/gravity/planet/registry/ -path '*tags/*' -type d | egrep 'tags/[^/]+$' | sort`
+- [ ] `sudo gravity system gc package --dry-run`
+
+Verify journal logs have been pruned [gravity.e#3429](https://github.com/gravitational/gravity.e/issues/3429)
+ - [ ] `sudo du -h /var/lib/gravity/planet/log/journal` should show only one subdir.
+
+Execute `gravity gc`.
+
+- [ ] Verify the gc operation completes successfully.
+ - `sudo gravity plan` (`--operation-id` may be looked up from `gravity status` or `gravity resource get operations`)
+- [ ] Verify packages from the previous installation have been removed locally.
```

Comment:
I would also add `gravity package list` command as a verification step that local packages have been removed.

Reply:
Thanks!  I was looking for pretty much this exact command during my testing.  I didn't find it, as my suggestion of `gravity system gc --dry-run` indicates.

I'd argue this points to a pretty severe discoverability problem.  For instance, here are some of the places I did look:

https://gravitational.com/gravity/docs/#packaging
https://gravitational.com/gravity/docs/cluster/#garbage-collection (and the entirety of this page)
https://gravitational.com/gravity/docs/search.html?q=package

```
$ ./7.0.11/gravity --help 2>&1 | grep package
  system service stop <package>
  system service start <package>
  system service status <package>
  system service journal <package> [<arg>...]
  system gc package [<flags>]
    Prune unused packages.

$ ./7.0.11/gravity --help-long | grep package
  system service stop <package>
  system service start <package>
  system service status <package>
  system service journal <package> [<arg>...]
  system gc package [<flags>]
    Prune unused packages.
    --dry-run  Only list packages to remove w/o removing them
    --cluster  Whether to prune cluster packages
```

Are we intentionally hiding `gravity package` and sub commands from end users?

Reply:
Yeah, this issue with was brought up by @eldios on our last team meeting. We will reevaluate our approach to CLI discoverability. Commands like these are not supposed to be used by users in general, but our CLI should definitely have a some sort of "power" mode where all available commands are listed.

Reply:
> Thanks! I was looking for pretty much this exact command during my testing. I didn't find it, as my suggestion of `gravity system gc --dry-run` indicates.
> 
> I'd argue this points to a pretty severe discoverability problem. For instance, here are some of the places I did look:
> 
> https://gravitational.com/gravity/docs/#packaging
> https://gravitational.com/gravity/docs/cluster/#garbage-collection (and the entirety of this page)
> https://gravitational.com/gravity/docs/search.html?q=package
> 
> ```
> $ ./7.0.11/gravity --help 2>&1 | grep package
>   system service stop <package>
>   system service start <package>
>   system service status <package>
>   system service journal <package> [<arg>...]
>   system gc package [<flags>]
>     Prune unused packages.
> 
> $ ./7.0.11/gravity --help-long | grep package
>   system service stop <package>
>   system service start <package>
>   system service status <package>
>   system service journal <package> [<arg>...]
>   system gc package [<flags>]
>     Prune unused packages.
>     --dry-run  Only list packages to remove w/o removing them
>     --cluster  Whether to prune cluster packages
> ```
> 
> Are we intentionally hiding `gravity package` and sub commands from end users?

Yes. https://github.com/gravitational/gravity/issues/1856

Reply:
Added in 22ab5e6.